zebra-chain/src/transaction.rs

```diff
@@ -194,6 +194,7 @@ impl Transaction {
                     .flat_map(|joinsplit| joinsplit.nullifiers.iter()),
             ),
             // No JoinSplits
+            Transaction::V1 { .. } | Transaction::V5 { .. } => Box::new(std::iter::empty()),
             _ => Box::new(std::iter::empty()),
```

Comment:
Let's match exhaustively here:
```suggestion
            Transaction::V1 { .. } | Transaction::V2 {
                joinsplit_data: None,
                ..
            } | Transaction::V3 {
                joinsplit_data: None,
                ..
            } | Transaction::V4 {
                joinsplit_data: None,
                ..
            } | Transaction::V5 { .. } => Box::new(std::iter::empty()),
```